### ISSUE TITLE
The training of the copy_augmented Transformer model is very slow!

### ISSUE THREAD

**Issue by Al-Sabahi** [kakaobrain/helo_word#6]
I have tried to train the model with copy option on 4GPU (12GB each) but the training was very slow. I have tried tuning some hyperparameters ending up with almost the same issue. Am I missing something:  here are my settings:

python train.py --track 1 --train-mode pretrain --model copy --ngpu 4 

            model_config = f"--ddp-backend=no_c10d --arch copy_augmented_transformer " \
                f"--update-freq 4 --alpha-warmup 10000 --optimizer adam --lr {lr} " \
                f"--dropout {dropout} --max-tokens 3000 --min-lr '1e-09' --save-interval-updates 5000 " \
                f"--lr-scheduler inverse_sqrt --weight-decay 0.0001 --max-epoch {max_epoch} " \
                f"--warmup-updates 4000 --warmup-init-lr '1e-07' --adam-betas '(0.9, 0.98)' "

**Comment by yjchoe** [comment]
Hi @Al-Sabahi, 

Thanks for trying out our model! We'll need more information to determine what the issue might be. The pretraining step iterates over a lot of synthetically generated data and is supposed to take a while. On our system, pre-training for 5 epochs took a day or so using 8 P40 GPUs. 

How slow (e.g., how much time does it take per epoch/steps) is the training? What kind of GPUs are you using? Is the speed significantly different from the speed when training with the `--model base` option instead? It'd be also helpful if you provide the log and/or the training curve from running the command on your system.

**Comment by Al-Sabahi** [comment]
Hi @yjchoe Thank you so much for the remarkable work and for the nice reply. I am suspecting that the problem is in my side. Because I am in the middle of something else, I will double check later. I am closing this issue and if I find something, I will reopen it. Thanks again.
